src/components/QuranReader/index.tsx

```diff
@@ -38,18 +40,19 @@ const verseFetcher = async (input: RequestInfo, init?: RequestInit) => {
 
 const QuranReader = ({ initialData, chapter }: QuranReaderProps) => {
   const quranReaderStyles = useSelector(selectQuranReaderStyles);
+  const currentTranslations = useSelector(selectCurrentTranslations) as string[];
   const { data, size, setSize, isValidating } = useSWRInfinite(
-    (index) => {
-      // TODO: select the translation using the user preference
-      return makeVersesUrl(chapter.id, {
+    (index) =>
+      makeVersesUrl(chapter.id, {
         page: index + 1,
         wordFields: `verse_key, verse_id, page_number, location, ${quranReaderStyles.quranFont}`,
-      });
-    },
+        translations: joinStringArray(currentTranslations),
+      }),
     verseFetcher,
     {
       initialData: initialData.verses,
-      revalidateOnFocus: true,
+      revalidateOnFocus: false, // disable auto revalidation when window gets focused
```

Comment:
Thank you for disabling this, so this was generating two request :) 

While adding v2 fonts, page won't render on initial page load. But when I click on dev utility, it was triggering the api request and rendering the page.